packages/keystone/fields/EncryptedText/README.md

```diff
@@ -0,0 +1,103 @@
+        hooks: {
+            resolveInput({ existingItem }) {
+                const decryptedNonPublicData = UserCrypto.nonPublicData.decrypt(existingItem.nonPublicData)
+            }
+        },
+    },
+  },
+  hooks: {
+      afterChanges: async ({ updatedItem }) => {
+         const decryptedNonPublicData = UserCrypto.nonPublicData.decrypt(updatedItem.nonPublicData)
+      },   
+  }, 
+});
+```
+
+## Advanced usage
+If you want to configure EncryptionManager yourself
+
+```js
+const { EncryptionManager } = require('@open-condo/keystone/crypto/EncryptionManager');
+const conf = require('@open-condo/config')
+
+const UserCrypto = {
+    nonPublicData: new EncryptionManager({
+       versions: { 'versionId':  { algorithm: 'crypto algorithm', secret: 'your secret key' } },
+       encryptionVersionId: 'versionId'
+    })
+}
+
+keystone.createList('User', {
+  fields: {
+      email: {type: 'Text'},
+      nonPublicData: {
+          type: 'EncryptedText',
+          encryptionManager: UserCrypto.nonPublicData,
+      },
+  }
+});
+```
+
+## Config extends Text field config
+
+| Option              | Type                | Default                   | Description                               |
+|---------------------|---------------------|---------------------------|-------------------------------------------|
+| `encryptionManager` | `EncryptionManager` | `new EncryptionManager()` | EncryptionManager instance with versions. |
+
+### `encryptionManager`
+
+We are using EncryptionManager for switching between algorithms / secret keys. This class helps to read data, 
+encrypted with old keys / algorithms. New data is being stored with CipherManager._encryptionVersionId
+
+## GraphQL
+
+`EncryptedText` fields behave as strings (`Text` field). On create / update operations input value will be
+encrypted before storing in database and returning in response. For read operations field expects encrypted value.
+
+## Storage
+
+The value stored is string containing provided version, iv and, encrypted with provided algorithm and secret, provided value.
+#### Example: \<our string to mark text encrypted\>:\<version from CipherManager\>:\<crypted data\>:\<iv or other service info\>
+
+## Service information
+
+| Env                                                | Format                                                    | Description                                                       |
+|----------------------------------------------------|-----------------------------------------------------------|-------------------------------------------------------------------|
+| `DEFAULT_KEYSTONE_SYMMETRIC_ENCRYPTION_CONFIG`     | `{ [id: string]: { algorithm: string, secret: string } }` | Default versions                                                  |
+| `DEFAULT_KEYSTONE_SYMMETRIC_ENCRYPTION_VERSION_ID` | `string`                                                  | Default version id from default versions to encrypt new data with |
```

Comment:
Why we need a DEFAULT prefix for environments?

Reply:
And why we need a keystone prefix?

Reply:
As for me it’s DATA_ENCRYPION_VERSION_ID and DATA_ENCRYPION_CONFIG if you don’t want to use word DATABASE as prefix.